File: WebService/WebService/Models/Resident.cs
using System;
using System.Collections.Generic;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using WebService.Models.Bases;

namespace WebService.Models
{
    public class Resident : AModelWithID
    {
        [BsonElement("firstName")]
        public string FirstName { get; set; }

        [BsonElement("lastName")]
        public string LastName { get; set; }

        [BsonElement("picture")]
        public byte[] Picture { get; set; }

        [BsonElement("room")]
        public string Room { get; set; }

        [BsonElement("birthday")]
        public DateTime? Birthday { get; set; }

        [BsonElement("doctor")]
        public Doctor Doctor { get; set; }

        [BsonElement("tags")]
        public List<int> Tags { get; set; } = new List<int>();

        [BsonElement("music")]
        public List<MediaUrl> Music { get; set; } = new List<MediaUrl>();

        [BsonElement("tunes")]
        public List<MediaUrl> Tunes { get; set; } = new List<MediaUrl>();

        [BsonElement("videos")]
        public List<MediaUrl> Videos { get; set; } = new List<MediaUrl>();

        [BsonElement("images")]
        public List<MediaUrl> Images { get; set; } = new List<MediaUrl>();

        [BsonElement("colors")]
        public List<Color> Colors { get; set; }

        [BsonElement("lastRecordedPosition")]
        public ResidentLocation LastRecordedPosition { get; set; }

        [BsonElement("locations")]
        public List<ObjectId> Locations { get; set; } = new List<ObjectId>();
    }
}

File: WebService/WebService/ClientApp/app/models/resident.ts
export class Resident {
    id: string
    firstName: string
    lastName: string
    picture: any
    pictureUrl: any;
    room: string
    birthday: Date
    doctor: Doctor
    images: Images
    videos: Videos
    constructor() {
        this.images = new Images();
        this.id = "";
        this.firstName = "";
        this.lastName = "";
        this.room = "";
        this.picture = "";
        this.pictureUrl = "";
        this.birthday = new Date();
        this.doctor = new Doctor();
        this.videos = new Videos();
    }
}

export class Doctor {
    name: string
    phoneNumber: string
    constructor() {
        this.name = "";
        this.phoneNumber = "";
    }
}

export class Images {
    id: string
    url: string
    name: string

    constructor() {
        this.id = "";
        this.url = "";
        this.name = "";
    }
}

export class Videos {
    id: string
    url: string
    type: string
    name: string

    constructor() {
        this.id = "";
        this.url = "";
        this.type = "";
        this.name = "";
    }
}

export class ProfilePicture {
    id: string
    url: string
    type: string
    name: string

    constructor() {
        this.id = "";
        this.id = "";
        this.type = "";
        this.name = "";
    }
}